augur/dates/ambiguous_date.py

```diff
@@ -131,3 +148,38 @@ def assert_only_less_significant_uncertainty(self):
                 raise InvalidDate(self.uncertain_date,
                     "Month contains uncertainty, so day must also be uncertain."
                 )
+
+
+def get_bounds(min_max_year):
+    """Get exact date bounds based on given years."""
+    # This must be an iterable with at least one value.
+    assert len(min_max_year) > 0
+
+    if len(min_max_year) > 2:
+        raise InvalidYearBounds(f"The year bounds {min_max_year!r} must have only one (lower) or two (lower, upper) bounds.")
+
+    lower_year = int(min_max_year[0])
+
+    if len(min_max_year) == 2:
+        upper_year = int(min_max_year[1])
+    else:
+        upper_year = None
+
+    # Ensure years are properly ordered.
+    if lower_year and upper_year and lower_year > upper_year:
+        lower_year, upper_year = upper_year, lower_year
+
+    try:
+        lower_bound = datetime.date(lower_year, 1, 1)
+    except:
+        raise InvalidYearBounds(f"{lower_year} is not a valid year.")
```

Comment:
_non-blocking_

I'm always a little wary of catch-all `except` clauses with handlers that assume a specific error, as they can lead to confusing situations. I'd be inclined to make the handler more specific. Maybe something like:

```suggestion
    try:
        lower_bound = datetime.date(lower_year, 1, 1)
    except ValueError as err:
        if str(err).startswith("year"):
            raise InvalidYearBounds(f"{lower_year} is not a valid year.") from err
        else:
            raise
```

Might also be ok to simplify by dropping the `startswith()` check and assuming all `ValueError`'s are about year. Better than assuming all errors are.